com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ShaderProgramBuilder.java

```diff
@@ -302,13 +302,20 @@ static public ShaderDesc.ResourceBinding.Builder SPIRVResourceToResourceBindingB
         ShaderDesc.ResourceType.Builder typeBuilder = getResourceTypeBuilder(res.type);
         resourceBindingBuilder.setType(typeBuilder);
         resourceBindingBuilder.setName(res.name);
-        resourceBindingBuilder.setNameHash(MurmurHash.hash64(res.name));
+        resourceBindingBuilder.setNameHash(res.nameHash);
+        resourceBindingBuilder.setId(res.id);
```

Comment:
ID is used in the editor because of how the shaderc pipeline mangles resource names. If we have something like:

```
uniform my_uniform_buffer {
  vec4 thing;
  vec4 bong;
};
```

it will translate into something like:

```
struct my_uniform_buffer
{
 // ..
};
uniform my_uniform_buffer _14; // 14 is the ID here
```

The opengl reflection will now give us uniform names as:
_14.thing;
_14.bong;

So we need a way to address these uniforms, which is where the ID comes in.